packages/binary/messagelayer/messagerequester/messagerequester.go

```diff
@@ -28,14 +27,7 @@ func New(missingMessages []message.ID, optionalOptions ...Option) *MessageReques
 	requester := &MessageRequester{
 		scheduledRequests: make(map[message.ID]*time.Timer),
 		options:           newOptions(optionalOptions),
-		Events: Events{
-			SendRequest: events.NewEvent(func(handler interface{}, params ...interface{}) {
-				handler.(func(message.ID))(params[0].(message.ID))
-			}),
-			MissingMessageAppeared: events.NewEvent(func(handler interface{}, params ...interface{}) {
-				handler.(func(message.ID))(params[0].(message.ID))
-			}),
-		},
+		Events:            *newEvents(),
```

Comment:
Okay it seems that sometimes `Events` are a pointer and sometimes not... Maybe make it consistently a pointer?